permission_handler_apple/ios/Classes/strategies/BluetoothPermissionStrategy.m

```diff
@@ -23,23 +24,28 @@ - (void)initManagerIfNeeded {
 }
 
 - (PermissionStatus)checkPermissionStatus:(PermissionGroup)permission {
-    [self initManagerIfNeeded];
-    if (@available(iOS 13.1, *)) {
+    if (@available(iOS 13.0, *)) {
         CBManagerAuthorization blePermission = [_centralManager authorization];
         return [BluetoothPermissionStrategy parsePermission:blePermission];
-    } else if (@available(iOS 13.0, *)){
-        CBManagerAuthorization blePermission =  [_centralManager authorization];
-        return [BluetoothPermissionStrategy parsePermission:blePermission];
     }
     return PermissionStatusGranted;
 }
 
-- (ServiceStatus)checkServiceStatus:(PermissionGroup)permission {
+- (void)checkServiceStatus:(PermissionGroup)permission completionHandler:(ServiceStatusHandler)completionHandler {
     [self initManagerIfNeeded];
+    
+    _serviceStatusHandler = completionHandler;
+    _requestedPermission = permission;
+}
+
+- (void)handleCheckServiceStatusCallback {
     if (@available(iOS 10, *)) {
-        return [_centralManager state] == CBManagerStatePoweredOn ? ServiceStatusEnabled : ServiceStatusDisabled;
+        ServiceStatus serviceStatus = [_centralManager state] == CBManagerStatePoweredOn ? ServiceStatusEnabled : ServiceStatusDisabled;
```

Comment:
I am wondering if this really covers all situations. If the `[_centralManager state]` returns `[CBManagerStateUnauthorized]` we could argue that the service is enabled but the application is not authorised. Unfortunately the [documentation](https://developer.apple.com/documentation/corebluetooth/cbmanagerstate?language=objc) is not really clear on this. 

Did you by change test this situation specifically?

Reply:
I also cannot find any documentation on `CBManagerStateUnauthorized`. Currently, I'm receiving `CBManagerStatePoweredOn` and `CBManagerStatePoweredOff`, when Bluetooth is enabled/disabled, respectively, regardless of Bluetooth permission.

We could opt for considering `CBManagerStateUnauthorized` as `ServiceStatusEnabled`. What do you think?

Reply:
I have tested the behavior. It works as follows:

- If the application has Bluetooth permission, everything works. We receive either `CBManagerStatePoweredOn` or `CBManagerStatePoweredOff`, depending on the Bluetooth service status.
- If the application was explicitly not granted access to Bluetooth (permission denied), we receive `CBManagerStateUnauthorized`, regardless of the Bluetooth service status.
- If the application has not requested Bluetooth permission yet, and therefore the permission state is unkown, the application will ask for permission first, and then carry out the logic as described in the two points above.

It seems that the last point is only true since iOS 13. https://support.apple.com/en-us/HT210578: 'Starting in iOS 13, iPadOS 13, watchOS 6, and tvOS 13, an app must ask permission to use Bluetooth functions'.

So basically, we can only reliably determine the Bluetooth status if permission is granted. Additionally, even checking the Bluetooth service status will prompt the user for permission.

What do you think, @mvanbeusekom? Would this be okay for the feature, provided we document this behavior?

Reply:
Agreed, if we document this clearly this should be fine.